npc_gzip/knn_compressor.py

```diff
@@ -0,0 +1,322 @@
+from typing import Sequence, Union
+
+import numpy as np
+from tqdm import tqdm
+
+from npc_gzip.compressors.base import BaseCompressor
+from npc_gzip.distance import Distance
+from npc_gzip.exceptions import (
+    InputLabelEqualLengthException,
+    InvalidObjectTypeException,
+    UnsupportedDistanceMetricException,
+)
+
+
+class KnnCompressor:
```

Comment:
Would be better to change the name as `KnnCompressor` is a bit confusing.

Reply:
Yep talked offline, updated to KnnClassifier